### PR TITLE
Migrate from Travis to Github Actions

### PR DESCRIPTION
**Pull request by ChristianSassine**
Replaces travis CI with GHA.

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+name: Erlang CI
+on: push
+jobs:
+  Test:
+    strategy:
+      matrix:
+        erlang: ['24']
```
Comment (berbiche) [inline .github/workflows/ci.yml]:
```suggestion
        erlang: ['24', '26']
```
We are almost done with the migration to OTP 26, so let's start testing OTP 26 too.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        env:
+          cache-name: rebar3
+        with:
+          path: |
+            _build
+            ~/.cache/rebar3
+          key: ci-${{runner.os}}-${{env.cache-name}}-erlang_${{matrix.erlang}}-rebar3-${{hashFiles('rebar.lock')}}
+          restore-keys: |
+            ci-${{runner.os}}-${{env.cache-name}}-erlang_${{matrix.erlang}}-rebar3
+            ci-${{runner.os}}-${{env.cache-name}}-erlang_${{matrix.erlang}}
```
Comment (berbiche) [inline .github/workflows/ci.yml]: From my testing, the restore-keys logic coupled with our setup makes it so upgrading the lockfile will make the CI keep using stale dependencies from the cache.
```suggestion
```

**Review by berbiche — APPROVED** [review]
LGTM overall.
Could you drop the linting/formatting changes?
They don't follow the standards we use in other Erlang projects.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (ChristianSassine) [reply .github/workflows/ci.yml]: Nice catch! Looking for a backup cache while the dependencies have changed would be counter-intuitive. Fixed.

**Pull request merged**